src/test/java/com/gooddata/schedule/ScheduleTest.java

```diff
@@ -0,0 +1,136 @@
+
+    @Mock
+    private DataloadProcess process;
+
+    @BeforeMethod
+    public void setUp() throws Exception {
+        MockitoAnnotations.initMocks(this);
+
+        when(process.getExecutables()).thenReturn(Collections.singleton(EXECUTABLE));
+        when(process.getId()).thenReturn("process_id");
+        doThrow(new IllegalArgumentException("wrong executable")).when(process).validateExecutable(Matchers.argThat(not(is(EXECUTABLE))));
+    }
+
+    @Test
+    public void testDeserialization() throws Exception {
+        final Schedule schedule = new ObjectMapper()
+                .readValue(getClass().getResourceAsStream("/schedules/schedule.json"), Schedule.class);
+
+        assertThat(schedule, is(notNullValue()));
+        assertThat(schedule.getId(), is("SCHEDULE_ID"));
+        assertThat(schedule.getType(), is("MSETL"));
+        assertThat(schedule.getState(), is("ENABLED"));
+        assertThat(schedule.isEnabled(), is(true));
+        assertThat(schedule.getCron(), is("0 0 * * *"));
+        assertThat(schedule.getTimezone(), is("UTC"));
+        assertThat(schedule.getConsecutiveFailedExecutionCount(), is(0));
+        assertThat(schedule.getProcessId(), is("process_id"));
+        assertThat(schedule.getExecutable(), is(EXECUTABLE));
+        assertThat(schedule.getNextExecutionTime(), is(DateTime.parse("2013-11-16T00:00:00.000Z")));
+        assertThat(schedule.getUri(), is("/gdc/projects/PROJECT_ID/schedules/SCHEDULE_ID"));
+    }
+
+    @Test
+    public void testSerialization() {
+        final Schedule schedule = new Schedule(process, EXECUTABLE, "0 0 * * *", "UTC");
+        assertThat(schedule, serializesToJson("/schedules/schedule-input.json"));
+    }
+
+    @Test
+    public void testIncorrectConstructorParams() {
+        try {
+            new Schedule(null, EXECUTABLE, "0 0 * * *", "UTC");
+            fail("IllegalArgumentException must be thrown");
+        } catch (IllegalArgumentException expected) {
+            assertThat(expected.getMessage(), containsString("process"));
+        }
+        try {
+            new Schedule(process, "garbage", "0 0 * * *", "UTC");
+            fail("IllegalArgumentException must be thrown");
+        } catch (IllegalArgumentException expected) {
+            assertThat(expected.getMessage(), containsString("wrong executable"));
+        }
+        try {
+            new Schedule(process, EXECUTABLE, "", "UTC");
+            fail("IllegalArgumentException must be thrown");
+        } catch (IllegalArgumentException expected) {
+            assertThat(expected.getMessage(), containsString("cron can't be empty"));
+        }
+        try {
+            new Schedule(process, EXECUTABLE, "0 0 * * *", "");
+            fail("IllegalArgumentException must be thrown");
+        } catch (IllegalArgumentException expected) {
+            assertThat(expected.getMessage(), containsString("The datetime zone id '' is not recognised"));
+        }
+    }
+
+    @Test
```

Comment:
again, I would prefer to state the exception in `@Test` (including message matching) in simple case like this - it is more readable to me